atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -92,6 +94,46 @@ def __init__(self, config=None):
     def get_enabled_clusters_for_platform(self, platform):
         return self.cluster_configs.get(platform, [])
 
+    def get_odcs_config(self):
+        odcs_config = self.conf.get('odcs')
+        if odcs_config:
+            odcs_config = ODCSConfig(**odcs_config)
+        return odcs_config
+
+
+class ODCSConfig(object):
+    """
+    Configuration for ODCS integration.
+    """
+
+    def __init__(self, signing_intents, default_signing_intent):
+        self.default_signing_intent = default_signing_intent
+
+        self.signing_intents = []
+        for priority, intent in enumerate(reversed(signing_intents)):
```

Comment:
The priority is actually used extensively throughout the plugin. I don't think ordereddict is the right data structure here as we definitely don't want to modify it.

The reason for rearranging them is because they're are listed in reactor-config as such:
```
- release
- beta
- unsigned
```
In this case, `unsigned` is the least restrictive signing intent. If I don't reverse the list, the plugin would have to use `max` instead of `min` and that seems harder to understand because we'll be using `max` to find the least restrictive intent.

The alternative is to change the design to list the signing intents differently:
```
- unsigned
- beta
- release
```

The reason I didn't do that initially is because when someone is setting up the reactor-config, it seems more natural to write the most important signing intent first.

Reply:
Makes sense to me. Maybe add a comment explaining that they appear in the config in reverse order of priority (i.e. most restrictive first). Or maybe call it "restrictiveness" instead of "priority"?

Reply:
Correct, so `collections.OrderedDict` would help here - items can be appended and after that should be sorted by the key. Later on `self.signing_intents.popitem(last=False)` would be to get an item with the top priority.

There is nothing with current implementation though, it just needs a comment about `reversed`

Reply:
Changed to `restrictiveness` and added comment.